compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1126,9 +1127,32 @@ static SmallVector<int64_t> getLinalgExtDefaultWorkgroupTileSizes(
   return workgroupTileSizes;
 }
 
+static LogicalResult setTransformStrategyRootConfig(func::FuncOp entryPointFn,
```

Comment:
Can you please refactor the KernelDispatch for CPU similarly to what @ThomasRaoux did here?

https://github.com/openxla/iree/pull/13046

This way we can easily scale to more cases and put the switch on the TD side: if TD knows how to handle a particular case well (a case is a combination of matcher, sizes, dtypes, HW etc) then it takes it, otherwise, we pass through.